tests/python/test_ggui.py

```diff
@@ -387,3 +387,48 @@ def render():
     render()
     verify_image(window.get_image_buffer_as_numpy(), 'test_set_image')
     window.destroy()
+
+
+@pytest.mark.skipif(not _ti_core.GGUI_AVAILABLE, reason="GGUI Not Available")
+@test_utils.test(arch=supported_archs)
+def test_draw_lines():
+    ti.init(arch=ti.vulkan)
+    N = 10
+    particles_pos = ti.Vector.field(3, dtype=ti.f32, shape=N)
+    points_pos = ti.Vector.field(3, dtype=ti.f32, shape=N)
+
+    @ti.kernel
+    def init_points_pos(points: ti.template()):
+        for i in range(points.shape[0]):
+            points[i] = [i for j in ti.static(range(3))]
+
+    init_points_pos(particles_pos)
+    init_points_pos(points_pos)
+
+    window = ti.ui.Window("Test for Drawing 3d-lines", (768, 768),
+                          show_window=False)
+    canvas = window.get_canvas()
+    scene = ti.ui.Scene()
+    camera = ti.ui.make_camera()
+    camera.position(0, 5, -10)
+    camera.lookat(3, 3, 1)
+
+    def render():
+        scene.set_camera(camera)
+        scene.ambient_light((0.8, 0.8, 0.8))
+        scene.point_light(pos=(0.5, 1.5, 1.5), color=(1, 1, 1))
+
+        scene.particles(particles_pos, color=(0.68, 0.26, 0.19), radius=0.5)
+        scene.lines(points_pos, color=(0.28, 0.68, 0.99), width=5.0)
+        canvas.scene(scene)
+
+    for _ in range(RENDER_REPEAT):
+        render()
+        window.get_image_buffer_as_numpy()
+
+    render()
+    if (platform.system() == 'Darwin'):
+        verify_image(window.get_image_buffer_as_numpy(), 'test_draw_lines.mac')
```

Comment:
Can you add a comment explaining why the rendered result differs by platform? Thanks!

Reply:
Just because of platform difference, but I think it's a bug. I will leave a `TODO` for this.